File: Top Interview 150/C++ Code/Binary Search/153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& a) {
        int n=a.size();
        if(n==1) return a[0];
        if(a[0]<a[n-1]) return a[0];
        int l=0,h=n-1;
        while(l<=h){
            int mid=(l+h)/2;
           if(a[mid+1]<a[mid]) return a[mid+1];
            else  if(a[mid]<a[mid-1]) return a[mid];
            else if(a[0]<a[mid]) l=mid+1;
            else h=mid-1;
        }
        return 0;
    }
};

File: Top Interview 150/Java Code/Binary Search/153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length==1) return nums[0];
         int l = 0, r = nums.length - 1;
        if (nums[r] > nums[0]) return nums[0];
        while (l <= r) {
            int mid = (l + r) / 2;
            if (mid + 1 < nums.length && nums[mid + 1] < nums[mid]) return nums[mid + 1];
            else if (mid > 0 && nums[mid] < nums[mid - 1]) return nums[mid];
            else if (nums[mid] > nums[0]) l = mid + 1;
            else r = mid - 1;
        }
        return -1;
    }
}
